autometa/common/external/samtools.py

```diff
@@ -42,49 +42,62 @@ def sort(sam, bam, nproc=mp.cpu_count()):
         </path/to/alignment.sam>
     bam : str
         </path/to/output/alignment.bam>
-    nproc : int, optional
+    cpus : int, optional
         Number of processors to be used. By default uses all the processors of the system
 
     Raises
     ------
     TypeError
-        nproc must be an integer greater than zero
+        cpus must be an integer greater than zero
     FileNotFoundError
         Specified path is incorrect or the file is empty
-    ChildProcessError
-        Samtools did not run successfully, returns the return code from subprocessing call
+    subprocess.CalledProcessError
+        Samtools did not run successfully
     """
 
-    if type(nproc) is not int or nproc <= 0:
-        raise TypeError(f"nproc must be an integer greater than zero! Given: {nproc}")
-    if not os.path.exists(sam) or os.stat(sam).st_size == 0:
+    if type(cpus) is not int or cpus <= 0:
+        raise TypeError(f"cpus must be an integer greater than zero! Given: {cpus}")
+    if not os.path.exists(sam) or not os.path.getsize(sam):
         raise FileNotFoundError(
             f"The specified path: {sam} is either incorrect or the file is empty"
         )
-    samtools_out_dir = os.path.dirname(os.path.abspath(bam))
-    err = os.path.join(samtools_out_dir, "samtools.err")
-    out = os.path.join(samtools_out_dir, "samtools.out")
-    with tempfile.TemporaryDirectory() as tempdir:
-        temp_bam = os.path.join(tempdir, os.path.basename(bam))
-        cmd = (
-            f"samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {temp_bam}"
+    with tempfile.TemporaryDirectory() as tempdir:  # this will delete the temporary files even if program stops in between
+        view_fname = "view_" + os.path.basename(bam)
+        view_fpath = os.path.join(tempdir, view_fname)
+        sort_fpath = os.path.join(tempdir, os.path.basename(bam))
+        # See https://stackoverflow.com/questions/13332268/how-to-use-subprocess-command-with-pipes
+        cmd_view = [
+            "samtools",
+            "view",
+            "-@",
+            str(cpus),
+            "-bS",
+            sam,
+            "-o",
+            view_fpath,
+        ]
+        cmd_sort = [
+            "samtools",
+            "sort",
+            "-@",
+            str(cpus),
+            view_fpath,
+            "-o",
+            sort_fpath,
+        ]
```

Comment:
As far as I can tell, you are not piping the output of `samtools view ...` to `samtools sort ...`. This is not the same behavior as before.

Reply:
Yes, I specifically avoided piping as then we have to use `shell=True`, which posses a some security risks and is also discouraged.
https://docs.python.org/3/library/subprocess.html#security-considerations
https://stackoverflow.com/questions/13332268/how-to-use-subprocess-command-with-pipes

Reply:
You have linked to an example of how to pipe subprocess commands, but you did not implement this. Currently, you are not piping any of the output from view to sort. This should be performed as was previously implemented.